test/spec/modules/parrableIdSystem_spec.js

```diff
@@ -242,6 +247,133 @@ describe('Parrable ID System', function() {
         })
       });
     });
+
+    describe('third party cookie support', function () {
+      let logErrorStub;
+      let callbackSpy = sinon.spy();
+
+      beforeEach(function() {
+        logErrorStub = sinon.stub(utils, 'logError');
+      });
+
+      afterEach(function () {
+        callbackSpy.resetHistory();
+        removeParrableCookie();
+      });
+
+      afterEach(function() {
+        logErrorStub.restore();
+      });
+
+      describe('when getting tpcSupport from XHR response', function () {
+        let request;
+        let dateNowStub;
+        const dateNowMock = 1;
+        const tpcSupportTtl = 1;
+
+        before(() => {
+          dateNowStub = sinon.stub(Date, 'now').returns(dateNowMock);
+        });
+
+        after(() => {
+          dateNowStub.restore();
+        });
+
+        it('should pass tpcSupport: true to the callback', function () {
+          let { callback } = parrableIdSubmodule.getId(P_CONFIG_MOCK);
+          callback(callbackSpy);
+          request = server.requests[0];
+
+          request.respond(
+            200,
+            RESPONSE_HEADERS,
+            JSON.stringify({ eid: P_XHR_EID, tpcSupport: true, tpcSupportTtl })
+          );
+
+          expect(callbackSpy.lastCall.args[0]).to.deep.equal({
+            eid: P_XHR_EID,
+            tpcSupport: true,
+            tpcUntil: dateNowMock + tpcSupportTtl
```

Comment:
These two values should not be returned to Prebid's UserID system, along with the rest of our values. It's not entirely apparent that we [return all values in the cookie](https://github.com/prebid/Prebid.js/blob/d3c7731f4dcd8c946236afc54dc1d748a3c1c9d8/modules/parrableIdSystem.js#L267) presently. These additional fields we are storing are internal-only to our system.

The `parrableIdSubmodule.getId(config)` function returns an object with `{ callback, id }`. The `id` value is populated with the current value of the cookie and `callback` is a function that also accepts a callback. Prebid uses this callback function to ask us to refresh our value post-auction. They call the `callback` function and pass their own function to be notified when we're done with our async work. [We also pass the updated parrableId to their callback function.](https://github.com/prebid/Prebid.js/blob/d3c7731f4dcd8c946236afc54dc1d748a3c1c9d8/modules/parrableIdSystem.js#L251)

[Here is where our submodule is called](https://github.com/prebid/Prebid.js/blob/7dba60cce0c083321047746013663cb51af25895/modules/userId/index.js#L679) (first to get the cookie value)

[Here is where the userId system processes the `callback` we return from the first getId() call](https://github.com/prebid/Prebid.js/blob/7dba60cce0c083321047746013663cb51af25895/modules/userId/index.js#L410)

----

**What to do?**

When reading the cookie we can return these values separately from the `parrableId` concept we have in the frontend code here. What is returned by `readCookie()` is the `parrableId` object layout: `{ eid, ibaOptout, ccpaOptout }` so we could update the `readCookie()` response to look like `{ parrableId: { eid, ibaOptout, ccpaOptout }, tpcSupport }` - no need to return the tpcUntil here if the date comparison (and subsequent suppression of the tpcSupport value if the date has passed(ttl expired)).

Reply:
So we shouldn't return tpcSupport at all. What's the point?

Reply:
We need to decide how we want to structure this since now what we read from the cookie is not what we return, but it was before this story. Maybe readCookie should return two objects: `parrableId` that contains eid, optouts, etc... and `internalParams` that contains the tpcSupport and tpcFilter for the other PBID story.